### PR TITLE
MAST: Tesscut query by objectname

### PR DESCRIPTION
**Pull request by ceb8**
This PR encompasses one piece of new functionality and a number of small improvements and bugfixes:

- **Tesscut:** Added the ability to query by objectname, where objectname can be a TIC ID, or something like "M101." 
The objectname argument works just like it does in the Observastions/Catalogs classes, where the given objectname is passed through the MAST resolver to get the correct RA/Dec to query.
The argument was added at the end of the argument list to preserve backwards compatibility.
Added new unit tests and documentation to go along with this new functionality.
- **resolve_object:** Make public function. 
I deprecated _resolve_object, replacing it with the public function resolve_object, in response to user requests for this functionality.
- **filter_products:** Bugfix on extension filter
The "extension" filter was not working because the function assumed it was given as an array, while in fact an array or a string should both be valid inputs.
- **TIC query criteria with position:** bugfix.
Some stray commas were breaking this function.
- **Panstarrs tests:** Various small bugfixes on the panstarrs tests

### CODE REVIEW

File: astroquery/mast/tesscut.py [hunk 1]
```diff
@@ -48,16 +86,21 @@ def __init__(self):
...
         with the given search area.
 
         Parameters
         ----------
-        coordinates : str or `astropy.coordinates` object
+        coordinates : str or `astropy.coordinates` object, optional
             The target around which to search. It may be specified as a
             string or as the appropriate `astropy.coordinates` object.
+            One and only one of coordinates and objectname must be supplied.
+        objectname : str, optional
```
Comment (bsipocz) [inline astroquery/mast/tesscut.py]: Could we keep the order of the parameters the same as in the signature?

File: astroquery/mast/core.py [hunk 2]
```diff
@@ -831,14 +831,23 @@ def _check_catalogs_criteria_params(self, criteria):
 
         return criteria_check
 
+    @deprecated(since="v0.3.10", alternative="resolve_object")
```
[the remark this answers is not in the document]
Comment (ceb8) [reply astroquery/mast/core.py]: So just delete this function completely? I can do that.

File: astroquery/mast/tesscut.py (hunk 1 above)
Comment (ceb8) [reply astroquery/mast/tesscut.py]: I can do that. I wasn't sure what the best way to handle it was. I think the order in the signature is awkward but necessary to allow for backward compatibility, but went ahead and documented the parameters in the logical order that groups coordinates/objectname together. But maybe consistency is more important in this instance.

Comment (bsipocz) [reply astroquery/mast/tesscut.py]: yes, I agree that it's all way to awkward. I hope once we switch off from python2 we can start using keyword only arguments and then we finally be able to reshuffle the signature the way it's most natural (well, do that at least for new modules, we would still have the issue with backward compatibility for the old ones).

File: astroquery/mast/core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/mast/core.py]: yes, I think that's reasonable. Rule of thumb, if something is not shared as public API (e.g. not in the docs) then it's fair game to remove. Especially when they even start with an underscore.

[the remark this answers is not in the document]
Comment (ceb8) [reply astroquery/mast/core.py]: That makes sense. Done.

**Comment by bsipocz** [comment]
Thanks @ceb8!

**Pull request merged**